### DISCUSSION TITLE
Reactive kudos

### DISCUSSION THREAD

**Discussion by jshaffstall** [anvilistas/anvil-labs#149] [General]
I know I'm really late to trying out the atomic module.  I'd intended to do it back when it was announced (three years ago now?), but I finally got around to using it in a small project to see how it worked.

I have to say it's amazing!  It simplifies so much, and really brings me back into MVC territory, where most of my Anvil apps are honestly a bit of a mess no matter how much I try to keep them from getting that way.   

Keeping global client state as an atom makes updating anything that depends on that trivial.  Making login/logout buttons visible invisible based on whether the user is logged in or out just automatically works everywhere.  

And it even plays well with @property decorators.

I don't think it's worth rewriting my big projects to use it (too much legacy code there to refactor at this point), but I think I'll be using it in all my projects going forward.  I'm hoping it's still something that's in consideration to put into Anvil Extras at some point.

**Comment by s-cork** [comment]
I think the proposal has moved to the anvil reactive implementation

see this forum post:
https://anvil.works/forum/t/anvil-reactive-add-reactivity-to-your-anvil-apps/19526

here's the repo:
https://github.com/anvilistas/reactive

Third Party app id:
N7KFE4YBWMGWJ5OX

This should bring some improvements and simplifies things somewhat.
We'll probably stick with it as a standalone repo for now.
It's quite a different way to think about your anvil state.

**Comment by jshaffstall** [comment]
I changed the title to reflect that reactive is getting the kudos.  It's definitely easier to use than atomic, and gives the same MVC feel.  

So much boilerplate just goes away with reactive.  Having a global state is super easy, stocking up changes until the user wants to push them to the server is easy, caching infrequently changing data in the client is easy, etc.  

So far I've only used reactive_class, render_effect, and effect.  I truly appreciate the magic that goes into making all those work, and work efficiently.
